hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/DefaultAMSProcessor.java

```diff
@@ -351,7 +351,19 @@ public void allocate(ApplicationAttemptId appAttemptId,
         ((AbstractYarnScheduler)getScheduler())
             .getApplicationAttempt(appAttemptId).pullUpdateContainerErrors());
 
-    response.setNumClusterNodes(getScheduler().getNumClusterNodes());
+    String label="";
+    try {
+      label = rmContext.getScheduler()
+          .getQueueInfo(app.getQueue(), false, false)
+          .getDefaultNodeLabelExpression();
+    } catch (Exception e){
+    }
```

Comment:
One thing about this: is the "queue missing" is a valid scenario here? If it is, and it's completely normal to not have any queue returned here, can you please write a comment (or a debug log line)? If it's not a valid scenario an exception could be thrown.

Reply:
Reference the SchedulerUtils. NormalizeAndValidateRequest fuction
`
try {
        queueInfo = rmContext.getScheduler().getQueueInfo(queueName, false,
            false);
      } catch (IOException e) {
        //Queue may not exist since it could be auto-created in case of
        // dynamic queues
      }
`
Should I add comments like this too？

Reply:
@brumi1024  hi，Complete the test case ！